src/battle/battle_script_commands.c

```diff
@@ -1646,6 +1716,9 @@ u32 CalculateBallShakes(void *bw, struct BattleStruct *sp)
 
 #endif
     }
+    if(sp->item_work = ITEM_FRIEND_BALL && i > 4){
+        SetMonData(GetPartyMon(bw,sp->defence_client,0),9,200);
+    }
```

Comment:
needs 
```c
if (sp->item_work == ITEM_FRIEND_BALL && i >= 4)
{
    SetMonData(GetPartyMon(bw,sp->defence_client,0), ID_PARA_friend, 200);
}
```
also needs to be handled for captures which are certain to pass (the if... else right above this)

Reply:
Added a copy of this section without `&& i >= 4` to the end of the critical captures shake count calculation to catch single shake captures.